packages/backend/src/managers/playground/McpServerManager.spec.ts

```diff
@@ -0,0 +1,124 @@
+    consoleErrors = '';
+    vi.spyOn(console, 'error').mockImplementation((...args) => {
+      consoleErrors += args.join(' ') + '\n';
+    });
+    consoleWarnings = '';
+    vi.spyOn(console, 'warn').mockImplementation((...args) => {
+      consoleWarnings += args.join(' ') + '\n';
+    });
+  });
+  afterEach(async () => {
+    await fs.promises.rm(appUserDirectory, { recursive: true });
+  });
+  describe('load', () => {
+    test('with no file, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toContain('McpServerManager: Failed to load MCP settings');
+      expect(consoleWarnings).toBe('');
+    });
+    test('with empty file, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      vi.spyOn(fs.promises, 'readFile').mockImplementation(async () => '{}');
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toBe('');
+      expect(consoleWarnings).toBe('');
+    });
+    test('with invalid JSON, returns empty array', async () => {
+      const mcpServerManager = new McpServerManager(appUserDirectory);
+      vi.spyOn(fs.promises, 'readFile').mockImplementation(async () => '{invalid json}');
+      const mcpServers = await mcpServerManager.load();
+      expect(mcpServers).toEqual([]);
+      expect(consoleErrors).toContain('McpServerManager: Failed to load MCP settings');
+      expect(consoleWarnings).toBe('');
+    });
+    describe('with valid JSON', () => {
+      let mcpServers: McpServer[];
+      beforeEach(async () => {
+        const mcpSettings = {
+          servers: {
+            'stdio-ok': {
+              enabled: true,
+              type: 'stdio',
+              command: 'npx',
+              args: ['-y', 'kubernetes-mcp-server'],
+            },
+            'sse-ok': {
+              enabled: true,
+              type: 'sse',
+              url: 'https://echo.example.com/sse',
+              headers: {
+                foo: 'bar',
+              },
+            },
+            'invalid-type': {
+              enabled: true,
+              type: 'invalid',
+              url: 'https://echo.example.com/sse',
+            },
+          },
+        };
+        vi.spyOn(fs.promises, 'readFile').mockImplementation(async () => JSON.stringify(mcpSettings));
+        mcpServers = await new McpServerManager(appUserDirectory).load();
+      });
+      test('parses stdio server', async () => {
+        const stdioOk: McpServer | undefined = mcpServers.find(server => server.name === 'stdio-ok');
+        expect(stdioOk).toEqual({
+          name: 'stdio-ok',
+          enabled: true,
+          type: McpServerType.STDIO,
+          command: 'npx',
+          args: ['-y', 'kubernetes-mcp-server'],
+        });
```

Comment:
You can use [expect.arrayContaining(array)](https://jestjs.io/docs/expect#expectarraycontainingarray)